recipes/hyperscan/all/conanfile.py

```diff
@@ -0,0 +1,75 @@
+from conans import ConanFile, CMake, tools
+
+import os
+
+class HyperscanConan(ConanFile):
+    name = "hyperscan"
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.hyperscan.io"
+    description = "High-performance regular expression matching library"
+    topics = ("regex", "regular expressions")
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    _source_subfolder = "source_subfolder"
+
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "optimise": [True, False],
+        "debug_output": [True, False],
+        "build_avx512": [True, False],
+        "fat_runtime": [True, False],
+        "build_chimera": [True, False],
+        "dump_support": [True, False]
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "optimise": True,
+        "debug_output": False,
+        "build_avx512": False,
+        "fat_runtime": False,
+        "build_chimera": False,
+        "dump_support": False
+    }
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("hyperscan-{0}".format(self.version), self._source_subfolder)
+
+    def build_requirements(self):
+        self.build_requires("boost/1.73.0");
```

Comment:
This is an odd build requirement :thinking:

Reply:
hyperscan uses some boost libs that are header only (graph, iterator, ...). I only found best practices to create recipes for header only libraries, but nothing about how to consume them.

Reply:
It's a build requirement because it hides it's dependencies within the `*.c` and are not exposed to it's consumer which is very nice of them